oak-run/src/main/java/org/apache/jackrabbit/oak/run/GenerateGarbageCommand.java

```diff
@@ -0,0 +1,427 @@
+            System.err.println("generateFullGC mode only available for DocumentNodeStore");
+            System.exit(1);
+        }
+
+        String generationBasePath = GEN_BASE_PATH + timestamp + "_" + runIndex;
+        System.out.println("Generating fullGC on the document: " + generationBasePath);
+        documentNodeStore = builder.build();
+
+        //VersionGarbageCollector.FullGCMode fullGCMode = getFullGCMode(options);
+        if (GenerateGarbageHelper.isInvalidGarbageGenerationMode(options.getGarbageType())) {
+            LOG.error("Invalid fullGCMode specified: " + options.getGarbageType() + " in: " + getClass().getName());
+            System.exit(1);
+        }
+
+        //1. Create nodes with properties
+        NodeBuilder rootNode = documentNodeStore.getRoot().builder();
+        NodeBuilder garbageRootNode = rootNode.child(GARBAGE_GEN_ROOT_PATH_BASE).child(GARBAGE_GEN_ROOT_NODE_NAME);
+        garbageRootNode.child(generationBasePath).setProperty(JcrConstants.JCR_PRIMARYTYPE, NodeTypeConstants.NT_OAK_UNSTRUCTURED, NAME);
+
+        int nodesCountUnderParent = options.getCreateGarbageNodesCount() / options.getGarbageNodesParentCount();
+        for(int i = 0; i < options.getGarbageNodesParentCount(); i ++) {
+            garbageRootNode.child(generationBasePath).child(GEN_PARENT_NODE_PREFIX + i).setProperty(JcrConstants.JCR_PRIMARYTYPE, "nt:folder", NAME);
+
+            for(int j = 0; j < nodesCountUnderParent; j ++) {
+                garbageRootNode.child(generationBasePath).child(GEN_PARENT_NODE_PREFIX + i).child(GEN_NODE_PREFIX + j).
+                        setProperty(JcrConstants.JCR_PRIMARYTYPE, NodeTypeConstants.NT_OAK_UNSTRUCTURED, NAME);
+
+                if (GenerateGarbageHelper.includesEmptyProps(options.getGarbageType())) {
+                    garbageRootNode.child(generationBasePath).child(GEN_PARENT_NODE_PREFIX + i).child(GEN_NODE_PREFIX + j).
+                            setProperty(EMPTY_PROPERTY_NAME, "bar", STRING);
+                }
+            }
+        }
+        documentNodeStore.merge(rootNode, EmptyHook.INSTANCE, CommitInfo.EMPTY);
+        documentNodeStore.runBackgroundOperations();
+
+
+        //2. Generate garbage nodes - EMPTY_PROPERTIES
+        if (GenerateGarbageHelper.includesEmptyProps(options.getGarbageType())) {
+            for (int i = 0; i < options.getGarbageNodesParentCount(); i++) {
+                for (int j = 0; j < nodesCountUnderParent; j++) {
+                    garbageRootNode.child(generationBasePath).child(GEN_PARENT_NODE_PREFIX + i).child(GEN_NODE_PREFIX + j).
+                            removeProperty(EMPTY_PROPERTY_NAME);
+                }
+            }
+        }
+        documentNodeStore.merge(rootNode, EmptyHook.INSTANCE, CommitInfo.EMPTY);
+        documentNodeStore.runBackgroundOperations();
+
+        //3.1. Generate garbage nodes - GAP_ORPHANS - remove parent nodes
+        if (GenerateGarbageHelper.includesGapOrphans(options.getGarbageType())) {
```

Comment:
One problem with this approach is that the more garbage types we add in the future, the more if/else statements we need to add.

This would make the code more complex, less readable, and very hard to understand.

How about we create `enums` for fullGcMode with `createGarbage` method and each mode would create its type of garbage?

This would ensure that for adding a new type of garbage, we don't need to update the base class, add a new implementation of garbage type, and plug in the base class.

cc @stefan-egli @Joscorbe

Reply:
A sample implementation could be like this.

`private interface GarbageType {
        void createGarbage();
    }`

    `private class EmptyProps implements GarbageType {

        @Override
        public void createGarbage() {
            for (int i = 0; i < options.getGarbageNodesParentCount(); i++) {
                for (int j = 0; j < nodesCountUnderParent; j++) {
                    garbageRootNode.child(generationBasePath).child(GEN_PARENT_NODE_PREFIX + i).child(GEN_NODE_PREFIX + j).
                            removeProperty(EMPTY_PROPERTY_NAME);
                }
            }

            documentNodeStore.merge(rootNode, EmptyHook.INSTANCE, CommitInfo.EMPTY);
            documentNodeStore.runBackgroundOperations();
        }
    }`

Reply:
All these implementations could be saved inside a `Map` of `fullGcMode --> Implementation` and we can simply iterate over garbageType and create garbage of different types in one simple loop.

Reply:
Thank you for the suggestion, this is a good idea for the future, when we will have multiple garbageTypes generated by the GenerateGarbageCommand.
For now, since we only have 2 types, one if/else is short and quite readable.

As suggested, I have completely separated the methods for generating EMPTY_PROPS / GAP_ORPHANS garbage.

Will definitely revisit the rest of the suggestion when more garbageType generation methods are implemented.